services/workflows-service/src/storage/storage.controller.internal.ts

```diff
@@ -44,29 +46,38 @@ export class StorageControllerInternal {
       },
     },
   })
-  async uploadFile(@UploadedFile() file: Express.Multer.File) {
+  async uploadFile(@UploadedFile() file: Partial<Express.MulterS3.File>) {
     const id = await this.service.createFileLink({
-      fileNameOnDisk: file.filename,
+      uri: file.location || String(file.path),
+      fileNameOnDisk: String(file.path),
+      bucketKey: file.key,
       // Probably wrong. Would require adding a relationship (Prisma) and using connect.
       userId: '',
     });
 
     return { id };
   }
 
-  // curl -v http://localhost:3000/api/storage/1679322938093
-  @common.Get('/:id')
-  async getFileById(@Param('id') id: string, @Res() res: Response) {
+  // curl -v http://localhost:3000/api/storage/content/1679322938093
+  @common.Get('/content/:id')
+  async fetchFileContent(@Param('id') id: string, @Res() res: Response) {
     // currently ignoring user id due to no user info
-    const fileNameOnDisk = await this.service.getFileNameById({
+    const persistedFile = await this.service.getFileNameById({
       id,
       userId: '',
     });
-
-    if (!fileNameOnDisk) {
+    if (!persistedFile) {
```

Comment:
IMHO, it should be part of the `service`

Reply:
we've decided that's not part of the scope of the task,
as for the future, @alonp99 WDYT?

Reply:
I think validations (and checks in that sort) should be part of the service

Reply:
I think the file service has no knowledge of weather the file exists or not.
you provide him the link/configuration/action and it does whatever it needs to.
the file service should be de-coupled of database and controller logic as he has nothing to do with them.

what do you think?